Yosemite/Yosemite/Stores/CardPresentPaymentStore.swift

```diff
@@ -49,15 +51,24 @@ private extension CardPresentPaymentStore {
         // Over simplification. This is the point where we would receive
         // new data via the CardReaderService's stream of discovered readers
         // In here, we should redirect that data to Storage and also up to the UI.
-        // For now we are sending the data up to the UI after mapping CardReaderService.CardReader
-        // to Yosemite.CardReader.
-        cancellable = cardReaderService.discoveredReaders.sink { readers in
-            let yosemiteReaaders = readers.map {
-                Yosemite.CardReader(name: $0.name, serialNumber: $0.serial)
-            }
-
-            // This hurts a bit, but for now it works
-            completion(yosemiteReaaders)
-        }
+        // For now we are sending the data up to the UI directly
+        cardReaderService.discoveredReaders.sink { readers in
```

Comment:
So, just to be clear, the completion is passed the array of all readers discovered since discovery started (and not just the most recently discovered reader), right? Is there an order to [CardReader]? Will the completion be called only when a new element is added? Can the new element always be added `.last` ?

Reply:
> the completion is passed the array of all readers discovered since discovery started (and not just the most recently discovered reader), right? 

Correct.

>  Is there an order to [CardReader]?

Not at this point. I am not sure this would be the right place to order the readers anyway? How should we sort by? Should that be part of the presentation logic?

Reply:
Actually I just remembered that we are NOT going to implement "Go Fish" but, if multiple readers are detected we're displaying a list of them to choose from (I was actually scribbling some notes on this last night for the views) so...

Don't worry about ordering them at all. I'll be ordering them in the UI by signal strength.